modules/api/src/main/EventStream.scala

```diff
@@ -103,8 +103,11 @@ final class EventStream(
         case lila.challenge.Event.Create(c) if isMyChallenge(c) =>
           queue.offer(challengeJson("challenge")(c).some).unit
 
-        case lila.challenge.Event.Decline(c) if isMyChallenge(c) =>
-          queue.offer(challengeJson("challengeDeclined")(c).some).unit
+        case lila.challenge.Event.Decline(c, r) if isMyChallenge(c) =>
+          queue.offer((
+              challengeJson("challengeDeclined")(c) + ("reason" -> r.map(JsString(_)).getOrElse(JsNull) )
```

Comment:
Other events don't have a root-level `reason` property (only `type` and `challenge`). Is it ok to add this here, or would it be better to put something like a `declineReason` in `challenege`?